app/scripts/controllers/provisionedService.js

```diff
@@ -0,0 +1,96 @@
+'use strict';
+
+angular.module('openshiftConsole')
+  .controller('ProvisionedServiceController', function ($scope,
+                                                        $filter,
+                                                        $routeParams,
+                                                        DataService,
+                                                        ProjectsService) {
+    $scope.alerts = {};
+    $scope.displayName = null;
+    $scope.projectName = $routeParams.project;
+    $scope.serviceInstance = null;
+    $scope.serviceInstances = null;
+    $scope.serviceClasses = {};
+
+    $scope.breadcrumbs = [
+      {
+        title: "Provisioned Services",
+        link: "project/" + $routeParams.project + "/browse/service-instances"
+      },
+      {
+        title: $routeParams.instance
+      }
+    ];
+
+    var watches = [];
+
+    var setDisplayName = function() {
+      if(!$scope.serviceInstance || !$scope.serviceClasses) {
+        return;
+      }
+
+      $scope.displayName = $filter('serviceInstanceDisplayName')($scope.serviceInstance, $scope.serviceClasses);
+    };
+
+    var serviceResolved = function(service, action) {
+      $scope.loaded = true;
+      $scope.serviceInstance = service;
+
+      if (action === "DELETED") {
+        $scope.alerts["deleted"] = {
+          type: "warning",
+          message: "This provisioned service has been deleted."
+        };
+      }
+    };
+
+    ProjectsService
+      .get($routeParams.project)
+      .then(_.spread(function(project, context) {
+        $scope.project = project;
+        $scope.projectContext = context;
+
+        DataService
+          .get({
+            group: 'servicecatalog.k8s.io',
+            resource: 'instances'
+          }, $routeParams.instance, context, { errorNotification: false })
+          .then(function(service) {
+            serviceResolved(service);
+            setDisplayName();
+            watches.push(DataService.watchObject({
+              group: 'servicecatalog.k8s.io',
+              resource: 'instances'
+            }, $routeParams.instance, context, serviceResolved));
+
+          }, function(error) {
+            $scope.loaded = true;
+            $scope.alerts["load"] = {
+              type: "error",
+              message: "The service details could not be loaded.",
+              details: $filter('getErrorDetails')(error)
+            };
+          });
+
+        DataService.list({
+          group: 'servicecatalog.k8s.io',
+          resource: 'serviceclasses'
+        }, context, function(serviceClasses) {
+          $scope.serviceClasses = serviceClasses.by('metadata.name');
+          setDisplayName();
+        });
+
+        watches.push(DataService.watch({
```

Comment:
This watch can go since you're already watching the instance for this page above.